packages/format-library/src/link/index.js

```diff
@@ -55,23 +55,19 @@ function Edit( {
 			return;
 		}
 
-		resetClickTimeout();
+		resetClickCount();
 
-		// This timeout will be cleared and a new openedBy set if a new link is clicked
-		clickTimeout.current = setTimeout( () => {
-			openedBy.current = null;
-			clickTimeout.current = undefined;
-		}, 100 );
+		// The clickCount is used to determine if the link control is being closed by the onFocusOutside event.
+		clickCount.current = 1;
```

Comment:
This will set the clickCount to 1 on mount, as `addingLink` is false on mount. That means the toolbar button can't open the link control because the clickCount is "stuck" at 1.